src/store/slices/users/authService.ts

```diff
@@ -33,16 +37,68 @@ export const signInUser = createAsyncThunk(
   }
 );
 
+export const signInWithGoogle = createAsyncThunk(
+  'authService/signInWithGoogle',
+  async (data: LoginUserData): Promise<Partial<TLoginData>> => {
+    const userDataGoogle = {
+      loginData: {} as TUser,
+      isLoggedIn: false as boolean
+    };
+    await signInWithPopup(auth, provider)
+      .then((result: any) => {
+        const credential: any = GoogleAuthProvider.credentialFromResult(result);
+        const user = result.user;
+        const splitName = user.displayName.split(' ');
+        userDataGoogle.loginData = {
+          email: user.email,
+          uid: user.uid,
+          firstName: splitName[0],
+          lastName: splitName[1],
+          role: 'user',
+          auth: 'google',
+          emailVerified: user.emailVerified,
+          phoneNumber: user.phoneNumber,
+          photoURL: user.photoURL
+        };
+        userDataGoogle.isLoggedIn = true;
+        setUser(user.uid, userDataGoogle);
+        console.log(userDataGoogle);
+        return userDataGoogle;
+      })
+      .catch((error: any) => {
+        console.log(error.message);
+      });
+    return userDataGoogle;
+  }
+);
+
 export const registerUser = createAsyncThunk(
   'authService/registerUser',
-  async (formData: RegisterUserData): Promise<Partial<RegisterUserData | void>> => {
+  async (formData: RegisterUserData): Promise<Partial<TLoginData>> => {
+    const userDataEmail = {
+      loginData: {} as TUser,
+      isLoggedIn: false as boolean
+    };
     await createUserWithEmailAndPassword(auth, formData.email, formData.password)
-      .then(() => {
-        return formData;
+      .then((userCredential) => {
+        userDataEmail.loginData = {
+          email: userCredential.user.email,
+          uid: userCredential.user.uid,
+          emailVerified: userCredential.user.emailVerified,
+          phoneNumber: userCredential.user.phoneNumber,
+          photoURL: userCredential.user.photoURL,
+          firstName: formData.firstName,
+          lastName: formData.lastName,
+          role: 'user',
+          auth: 'email'
+        };
+        userDataEmail.isLoggedIn = true;
+        setUser(userCredential.user.uid, userDataEmail);
```

Comment:
make sure this is not triggered

Reply:
Checked, if the email is already registered in auth the setUser function does not trigger.